internal/services/eventgrid/eventgrid_partner_configuration_resource.go

```diff
@@ -0,0 +1,270 @@
+						AuthorizedPartnersList:             expandAuthorizedPartnersList(&config.PartnerAuthorizations),
+					},
+				},
+				Tags: pointer.To(config.Tags),
+			}
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerConfigurationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			id, err := commonids.ParseResourceGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config EventGridPartnerConfigurationResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("default_maximum_expiration_time_in_days") {
+				existing.Model.Properties.PartnerAuthorization.DefaultMaximumExpirationTimeInDays = pointer.To(config.DefaultMaximumExpirationTimeInDays)
+			}
+
+			if metadata.ResourceData.HasChange("partner_authorization") {
+				existing.Model.Properties.PartnerAuthorization.AuthorizedPartnersList = expandAuthorizedPartnersList(&config.PartnerAuthorizations)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				existing.Model.Tags = pointer.To(config.Tags)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerConfigurationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			id, err := commonids.ParseResourceGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := EventGridPartnerConfigurationResourceModel{
+				ResourceGroup: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.DefaultMaximumExpirationTimeInDays = *model.Properties.PartnerAuthorization.DefaultMaximumExpirationTimeInDays
+				state.PartnerAuthorizations = flattenAuthorizedPartnersList(model.Properties.PartnerAuthorization.AuthorizedPartnersList)
+
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+			}
```

Comment:
Many of these values can possible be `nil`, so we can check them first to avoid a panic.
```suggestion
			if model := resp.Model; model != nil {
				if props := model.Properties; props != nil {
					if auth := props.PartnerAuthorization; auth != nil {
						state.DefaultMaximumExpirationTimeInDays = pointer.From(auth.DefaultMaximumExpirationTimeInDays)
						state.PartnerAuthorizations = flattenAuthorizedPartnersList(auth.AuthorizedPartnersList)
					}
				}
				
				state.Tags = pointer.From(model.Tags)
			}
```